Doc/library/exceptions.rst

```diff
@@ -691,7 +691,9 @@ depending on the system error code.
 .. exception:: NotADirectoryError
 
    Raised when a directory operation (such as :func:`os.listdir`) is requested
-   on something which is not a directory.
+   on something which is not a directory.  Can also be raised on a file operation
+   that involves reading a directory, in which case this error may indicate that
+   path is either not a directory or does not exist.
```

Comment:
I'd emphasize that `NotADirectoryError` can be raised by a file operation that tries to open or traverse a non-directory file or symlink to a non-directory file as if it were a directory. The concrete actions "open" and "traverse" flesh out the more generic phrasing "directory operation". 

The case of a non-existing path component is unrelated. That should fail with `ENOENT`, i.e. `FileNotFoundError`. 

Here are some particular cases from POSIX that fail with `ENOTDIR`:

* A component of the path prefix (i.e. excluding the last component and trailing slash characters) names an existing file that is neither a directory nor a symbolic link to a directory. System calls: `open`,  `stat`, `mkdir`
* The path argument contains at least one non-slash character and ends with one or more trailing slash characters and the last pathname component names an existing file that is neither a directory nor a symbolic link to a directory. System calls: `open`, `stat`
* `O_DIRECTORY` was specified and the path argument resolves to a non-directory file. System calls: `open`
* A component of dirname names an existing file that is neither a directory nor a symbolic link to a directory. System calls: `opendir`

Reply:
I've misread the bug report in respect to non-existent path, thanks for catching that!